homeassistant/components/fritzbox_callmonitor/config_flow.py

```diff
@@ -0,0 +1,179 @@
+                CONF_USERNAME: self._username,
+                CONF_PASSWORD: self._password,
+                CONF_PHONEBOOK: self._phonebook_id,
+                CONF_PREFIXES: self._prefixes,
+            },
+        )
+
+    def _try_connect(self):
+        """Try to connect and check auth."""
+        self._phonebook = FritzBoxPhonebook(
+            host=self._host,
+            username=self._username,
+            password=self._password,
+            phonebook_id=self._phonebook_id,
+            prefixes=self._prefixes,
+        )
+
+        try:
+            self._phonebook.init_phonebook()
+            self._phonebook_ids = self._phonebook.get_phonebook_ids()
+            return RESULT_SUCCESS
+        except FritzConnectionException:
+            return RESULT_INVALID_AUTH
+        except requests.exceptions.ConnectionError:
+            return RESULT_NO_DEVIES_FOUND
+
+    def _is_already_configured(self, host, phonebook_id):
+        """Check if an entity with the same host and phonebook_id is already configured."""
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if (
+                entry.data[CONF_HOST] == host
+                and entry.data[CONF_PHONEBOOK] == phonebook_id
+            ):
+                return True
+
+    async def _get_name_of_phonebook(self, phonebook_id):
+        """Return name of phonebook for given phonebook_id."""
+        phonebook_info = await self.hass.async_add_executor_job(
+            partial(self._phonebook.fph.phonebook_info, phonebook_id)
+        )
+        return phonebook_info[FRITZ_ATTR_NAME]
+
+    async def async_step_import(self, user_input=None):
+        """Handle configuration by yaml file."""
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+
+            self._name = user_input[CONF_HOST]
+            self._host = user_input[CONF_HOST]
+            self._port = user_input[CONF_PORT]
+            self._password = user_input[CONF_PASSWORD]
+            self._username = user_input[CONF_USERNAME]
+            self._prefixes = user_input.get(CONF_PREFIXES)
+
+            if self._prefixes and self._prefixes.strip():
+                self._prefixes = [
+                    prefix.strip() for prefix in self._prefixes.split(",")
+                ]
+
+            result = await self.hass.async_add_executor_job(self._try_connect)
+
+            if result == RESULT_SUCCESS:
+                if len(self._phonebook_ids) > 1:
+                    return await self.async_step_phonebook()
+
+                self._phonebook_id = DEFAULT_PHONEBOOK
+                self._phonebook_name = await self._get_name_of_phonebook(
+                    self._phonebook_id
+                )
+
+                if self._is_already_configured(self._host, self._phonebook_id):
```

Comment:
Is there any reason we wouldn't just pull in all phone books and allow the user to disable the entities it doesn't need?

Reply:
Fair point, tbh did not really think about disabling unused phonebooks. I can certainly implement that all phonebooks are pulled. Would that be the preferred solution in most cases?

Reply:
You would put it all in config entry. But yes, you won't be able to configure per-phonebook settings easily anymore.

Reply:
I would prefer to leave it like this then, since it could be important for some users to have per-phonebook settings.